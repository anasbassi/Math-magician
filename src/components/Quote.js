import React from 'react';

const Quote = () => (
  <section className="quoteContainer">
    <p>
      Mathematics is not about numbers, equation, computations or algorithms:
      it is about understanding.- Willium Paul Thurston.
    </p>
  </section>
);

export default Quote;
