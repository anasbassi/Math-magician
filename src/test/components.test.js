import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/calculator';
import Home from '../components/Home';
import Quote from '../components/Quote';

describe('App component testing with snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Home />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Quote />);
    expect(tree).toMatchSnapshot();
  });

});
