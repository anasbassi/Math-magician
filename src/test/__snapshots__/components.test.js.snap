// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component testing with snapshot renders correctly 1`] = `
<section
  className="cal"
>
  <h2>
    Let's Do Some Maths
  </h2>
  <div
    className="calculator"
  >
    <div
      className="calculator-display"
      onChange={[Function]}
    >
      0
    </div>
    <div
      className="buttons"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="button zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;

exports[`App component testing with snapshot renders correctly 2`] = `
<section
  className="homeSection"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat.
  </p>
</section>
`;

exports[`App component testing with snapshot renders correctly 3`] = `
<section
  className="quoteContainer"
>
  <p>
    Mathematics is not about numbers, equation, computations or algorithms: it is about understanding.- Willium Paul Thurston.
  </p>
</section>
`;

exports[`App component testing with snapshot. renders correctly 1`] = `
<section
  className="cal"
>
  <h2>
    Let's Do Some Maths
  </h2>
  <div
    className="calculator"
  >
    <div
      className="calculator-display"
      onChange={[Function]}
    >
      0
    </div>
    <div
      className="buttons"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="button zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;

exports[`App component testing with snapshot. renders correctly 2`] = `
<section
  className="homeSection"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat.
  </p>
</section>
`;

exports[`App component testing with snapshot. renders correctly 3`] = `
<section
  className="quoteContainer"
>
  <p>
    Mathematics is not about numbers, equation, computations or algorithms: it is about understanding.- Willium Paul Thurston.
  </p>
</section>
`;
